# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ImageEncoder.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f dom/base/ImageEncoder.h
--- a/dom/base/ImageEncoder.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/dom/base/ImageEncoder.h	Wed Jun 10 22:21:02 2020 +0000
@@ -20,7 +20,7 @@
 namespace mozilla {
 
 namespace layers {
-class AsyncCanvasRenderer;
+class CanvasRenderer;
 class Image;
 }  // namespace layers
 
@@ -40,7 +40,7 @@
   static nsresult ExtractData(nsAString& aType, const nsAString& aOptions,
                               const nsIntSize aSize, bool aUsePlaceholder,
                               nsICanvasRenderingContextInternal* aContext,
-                              layers::AsyncCanvasRenderer* aRenderer,
+                              layers::CanvasRenderer* aRenderer,
                               nsIInputStream** aStream);
 
   // Extracts data asynchronously. aType may change to "image/png" if we had to
@@ -86,7 +86,7 @@
       const nsAString& aType, const nsAString& aOptions, uint8_t* aImageBuffer,
       int32_t aFormat, const nsIntSize aSize, bool aUsePlaceholder,
       layers::Image* aImage, nsICanvasRenderingContextInternal* aContext,
-      layers::AsyncCanvasRenderer* aRenderer, nsIInputStream** aStream,
+      layers::CanvasRenderer* aRenderer, nsIInputStream** aStream,
       imgIEncoder* aEncoder);
 
   // Creates and returns an encoder instance of the type specified in aType.